mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -24,6 +25,34 @@ def SYCL_Dialect : Dialect {
   let name = "sycl";
   let cppNamespace = "::mlir::sycl";
   let useDefaultTypePrinterParser = 1;
+  let extraClassDeclaration = [{
+    llvm::DenseMap<std::pair<::mlir::TypeID, llvm::StringRef>, llvm::StringRef>
```

Comment:
It is still unclear what this map is for. Looks like the key is the pair of TypeID and a StringRef. and the value another stringRef. Please explain with examples what this map will look like when populated. The pair <TypeID, StringRef> represents an indetifier for a SYCL type and the StringRef is the name of a method in that type ? Is the name mangled or unmangled ?  What is the value in the map supposed to be ? The unmangled name of the method. 

All of these questions should really be explained in a comment. 

Also, this looks to me like a 'registry' listing the methods that are available in a SYCL type. That information IMO does not belong inside the SYCL_Dialect.  I would suggest to place it in a separate class (e.g. SYCL_Registry).

Reply:
Will implement a registry class to hold the methods.